src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -5042,10 +5047,23 @@ namespace WaterThermalTanks {
 		Qnet = 0.0;
 		Qfuel = 0.0;
 
-		// Calculate steady-state heat rates
+		// Calculate the heating rate from the heat pump.
+		if ( Tank.HeatPumpNum > 0 ) {
+            HeatPumpWaterHeaterData const &HeatPump = HPWaterHeater(Tank.HeatPumpNum);
+			DataLoopNode::NodeData const &HPWHCondWaterInletNode = DataLoopNode::Node(HeatPump.CondWaterInletNode);
+			DataLoopNode::NodeData const &HPWHCondWaterOutletNode = DataLoopNode::Node(HeatPump.CondWaterOutletNode);
+			HPWHCondenserDeltaT = HPWHCondWaterOutletNode.Temp - HPWHCondWaterInletNode.Temp;
+		} else {
+			HPWHCondenserDeltaT = 0.0;
+		}
+        assert( HPWHCondenserDeltaT >= 0 );
```

Comment:
Hmmm, I suppose if the heat pump was attached to the tank backwards it could possibly be negative. Does the IDF construction allow that? Maybe I'll see if I can create an IDF that hits this error. If it is possible I'll change it to be a warning or error instead of assert.